AdobeStockImageAdminUi/view/adminhtml/templates/panel.phtml

```diff
@@ -5,16 +5,17 @@
  */
 ?>
 
-<div id="adobe-stock-images-search-modal" style="display: none;">
+<div id="adobe-stock-images-search-modal"
+     data-mage-init='{"Magento_Ui/js/modal/modal":{"type":"slide","title":"<?= $block->escapeJs(__('Adobe Stock')) ?>",
+         "buttons":[]}}' class="no-display">
     <div class="adobe-stock-images-search-modal-content">
         <?php echo $block->getChildHtml(); ?>
     </div>
 </div>
 
-<script type="text/x-magento-init">
-    {
-        "#adobe-stock-images-search-modal": {
-            "Magento_AdobeStockImageAdminUi/panel": {}
-        }
-    }
+<script>
+    require(['jquery', 'mage/apply/main'], function ($, main) {
```

Comment:
I'd avoid inline scripts if it's possible.

Reply:
Yes, I agree with you and @sidolov  regarding this pull request we need to avoid inline js and I don't think that this issue is with Magento 2 as gallery functionality will work without error in magento/magento2 repository. I think applyBinding() is triggered every time we open our Adobe stock modal so we need to fix this. I'm working on an alternative way with using the knockout js template to render the sliding modal. Let me know you have any other idea to fix this issue.

Thanks

Reply:
@konarshankar07 , we should fix gallery and make only one request when modal will be opened first time and reuse already loaded content in all cases, including the case when modal was closed and opened again.

Reply:
Yes, I will try to apply your logic. But can we implement differently without using the following code? I mean are there any other ways we can add our Uicomponent in the modal content.
https://github.com/magento/adobe-stock-integration/blob/7556d544db268309817c32e9cdc1ee8f98c9246b/AdobeStockImageAdminUi/view/adminhtml/web/panel.js#L18
`return function (config, element) {
        $(element).modal({
            type: 'slide',
            buttons: [],
            title: $t('Adobe Stock')
        }).applyBindings();
    };`

Reply:
@konarshankar07 I'm not sure that we have another valid way to add it, at least this one is more easiest. And we still have a bug when content loading more than once after modal init